### ISSUE TITLE
default halo finding options for member search

### ISSUE THREAD

**Issue by dnarayanan** [dnarayanan/caesar#12]
it looks like ```obj.member_search()``` works for a `simba` snapshot, but fails with a ```SystemExit: No Halo IDs found in snapshot -- please specify a source (haloid='fof' or 'snap')

/home/desika.narayanan/miniconda3/envs/pd3env/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3334: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.
  warn("To exit: use 'exit', 'quit', or Ctrl-D.", stacklevel=1)
```

if used on a non-`simba` snapshot.  i suspect this is because it is defaulting to `snap`.  this should probably default to `fof` so that it works in the more general sense by default, and has to be manually set for specific situations such as `simba`

**Comment by saethlin** [comment]
I think the ideal behavior is:

1. Obey the option if provided by the user. If unable to follow through with it, bail with a helpful message. This functionality basically already exists
2. If no option is provided by the user, attempt to load halo IDs from the snapshot. If such haloids are not available, run the FoF.

I think this can be achieved by just changing the last line of `load_haloid`. I'd break out the big `elif` for `'snap'` that's in there already, drop it in its own function, and call into that new function in the `elif` block that exists, and in the final line of `load_haloid`.

**Comment by dnarayanan** [comment]
hmmm i get worried about  'silent fails' like this - i.e. the user thinks there's haloIDs in their snapshot, but then the code actually ran fof.

**Comment by saethlin** [comment]
I agree. I see maybe three options?

1. Announce that everyone's code that runs caesar is broken, and they must now add an option to all their invocations that selects a source of haloIDs. I think this is pretty uncool, because it basically says everyone who isn't using SIMBA is a second-class citizen. 

2. Be as noisy as possible when defaulting to FoF to generate haloIDs. This is what I'd prefer, because it keeps everyone's existing code running. But if for whatever reason we want to incentivize people to use in-snapshot halo IDs (like if these are becoming the norm for simulations) it's not so great.

3. Offer some other means to turn on/off FoF as a fallback for haloIDs. This could be an install-time option (which would make it impossible for SIMBA users to forget about it) or I can also see it being an environment variable. This could be either a switch that SIMBA people flip on, or something that we demand everyone choose, but this way they'd only need to type something in to set it once.

**Comment by dnarayanan** [comment]
addressed by 145a7973215b3a9cef5488fc198f3c03d7463c71 and PR #14
